basil/dut.py

```diff
@@ -14,6 +12,12 @@
 import warnings
 from collections import OrderedDict
 
+# FIXME: Bad practice
+# Logger settings should not be defined in a module, but once by the
+# application developer. Thus outside of basil. Otherwise multiple calls to
+# the basic config are possible. This is left here at the moment for backward
+# compatibility and since our logging format is the same everywhere (?).
+import logging
```

Comment:
I find this comment ridiculous. Just saying something is bad practice that served us well for many years. Keep in mind that basil is still a hardware debugging and development framework.
Maybe your application and not the framework is doing something wrong. You still can import basil into your application and change the behavior of the global logger. Something that you should consider than changing the behavior for everyone else.

Reply:
This comment gives a reason why this should be removed in future and thus is a FIXME. What is the issue here?

Reply:
I give you an example why this is bad practice:

```python

from basil.dut import Dut

def my_func():
    logging.info('Bla')

if __name__ == '__main__':
    import logging
    logging.basicConfig(level=logging.INFO, format="My format %(message)s")
    my_func()
```

The own defined format is overwritten on basil.dut import. No think about several projects, each defining a logging format as basil does. The module import order would define the logging format.

Reply:
Just deleting the logging import is also a solution, but applications using basil development would suddenly not see any logs or logs formatted different (depending on their logging configuration). We could say it is development branch so we expect other project to comply. But this step I did not want to do here. But if you think that is better I'll just delete the lines.

Reply:
```PYTHON
import logging

# just set up the basil logger format and level here
logging.basicConfig(level=logging.ERROR)
from basil.dut import Dut

# Your own logger here
logger = logging.getLogger("test")
logger.propagate = False
logger.setLevel(logging.INFO)
console = logging.StreamHandler()
console.setLevel(logging.INFO)
formatter = logging.Formatter("My format %(message)s")
console.setFormatter(formatter)
logger.addHandler(console)

dut = Dut("")
dut.init()

def my_func():
    logging.error("logging info message")
    logger.info('logger info message')
    logger.error('logger error message')

if __name__ == '__main__':
    my_func()
```

Reply:
So you still want to try to fix the output in you application by changing the underlying framework. Good, but the above code snippet can be used in you application without changing basil.

Reply:
How do you set the logging of a basil module with your approach?